Update MLflow and Databricks docs (#14011)
langchain-ai/langchain: docs/docs/integrations/providers/databricks.md

                    "task": "llm/v1/chat",
                    "openai_config": {
                        "openai_api_key": "{{secrets/<scope>/<key>}}",
                    },
                },
            }
        ],
    },
)
chat = ChatDatabricks(endpoint=name, temperature=0.1)
print(chat([HumanMessage(content="hello")]))
# -> content='Hello! How can I assist you today?'
```

Databricks Foundation Model APIs
--------------------------------

[Databricks Foundation Model APIs](https://docs.databricks.com/machine-learning/foundation-models/index.html) which allow you to access and query state-of-the-art open source models from a dedicated serving endpoint. With Foundation Model APIs, developers can quickly and easily build applications that leverage a high-quality generative AI model without maintaining their own model deployment. The following example uses the `databricks-bge-large-en` endpoint to embed text:

```python
from langchain.llms import DatabricksEmbeddings

embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
print(embeddings.embed_query("hello")[:3])
# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
```

> Databricks Dolly

Author: Not sure if this is something we still want to highlight, but no harm to keep it.

Reviewer: Let's remove

Author: removed

Databricks MLflow makes it more convenient to develop LangChain applications on Databricks. For MLflow tracking, you don't need to set the tracking uri. For MLflow Model Serving, you can save LangChain Chains in the MLflow langchain flavor, and then register and serve the Chain with a few clicks on Databricks, with credentials securely managed by MLflow Model Serving.

Databricks as an LLM provider
-----------------------------

The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) illustrates the method to wrap Databricks endpoints as LLMs in LangChain. It supports two types of endpoints: the serving endpoint, which is recommended for both production and development, and the cluster driver proxy app, which is recommended for interactive development. 

> Databricks External Models

Author: @dbczumar Added `Databricks External Models` and `Databricks Foundation Model APIs`.

Databricks MLflow makes it more convenient to develop LangChain applications on Databricks. For MLflow tracking, you don't need to set the tracking uri. For MLflow Model Serving, you can save LangChain Chains in the MLflow langchain flavor, and then register and serve the Chain with a few clicks on Databricks, with credentials securely managed by MLflow Model Serving.

Databricks External Models
--------------------------

> [Databricks External Models](https://docs.databricks.com/generative-ai/external-models/index.html) is designed to streamline the usage and management of various large language model (LLM) providers, such as OpenAI and Anthropic, within an organization. It offers a high-level interface that simplifies the interaction with these services by providing a unified endpoint to handle specific LLM related requests. The following example creates an endpoint that serves OpenAI's GPT-4 and generates a chat response from it:

Reviewer 2: ```suggestion
[Databricks External Models](https://docs.databricks.com/generative-ai/external-models/index.html) is a service that is designed to streamline the usage and management of various large language model (LLM) providers, such as OpenAI and Anthropic, within an organization. It offers a high-level interface that simplifies the interaction with these services by providing a unified endpoint to handle specific LLM related requests. The following example creates an endpoint that serves OpenAI's GPT-4 model and generates a chat response from it:
```

                    "openai_config": {
                        "openai_api_key": "{{secrets/<scope>/<key>}}",
                    },
                },
            }
        ],
    },
)
chat = ChatDatabricks(endpoint=name, temperature=0.1)
print(chat([HumanMessage(content="hello")]))
# -> content='Hello! How can I assist you today?'
```

Databricks Foundation Model APIs
--------------------------------

[Databricks Foundation Model APIs](https://docs.databricks.com/machine-learning/foundation-models/index.html) which allow you to access and query state-of-the-art open source models from a dedicated serving endpoint. With Foundation Model APIs, developers can quickly and easily build applications that leverage a high-quality generative AI model without maintaining their own model deployment. The following example uses the `databricks-bge-large-en` endpoint to embed text:

```python
from langchain.llms import DatabricksEmbeddings

embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
print(embeddings.embed_query("hello")[:3])
# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
```

Databricks as an LLM provider
-----------------------------

> The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) illustrates the method to serve a custom model registered by MLflow as a Databricks endpoint.

Reviewer 2: ```suggestion
The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) demonstrates how to serve a custom model that has been registered by MLflow as a Databricks endpoint.
```

> The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks) illustrates the method to serve a custom model registered by MLflow as a Databricks endpoint.

Author: updated!

> The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks#wrapping-a-serving-endpoint-custom-model) demonstrates how to serve a custom model that has been registered by MLflow as a Databricks endpoint.

Author: checking this link is valid in the preview now.

> Databricks MLflow AI Gateway

Reviewer: @harupy Can we also update this page to show simpler completions examples for external / foundation model endpoints, and also show examples for chat & embeddings?

Author: `docs/docs/integrations/llms/databricks.ipynb` contains examples for external / foundation model endpoints. Can we use them?

Reviewer: This section is out of date: https://github.com/langchain-ai/langchain/blob/fc02bb88b1633cbd9f7290fd33aeef1091f29f63/docs/docs/integrations/providers/databricks.md#databricks-as-an-llm-provider

Author: Got it, I'll update them.
